src/sql/workbench/api/common/extHostModelViewDialog.ts

```diff
@@ -640,31 +660,41 @@ class ModelViewDashboardImpl implements azdata.window.ModelViewDashboard {
 		return this._editor.closeEditor();
 	}
 
-	createTab(tab: azdata.DashboardTab, view: azdata.ModelView): azdata.Tab {
-		if (tab.toolbar) {
+	createTab(dashboardTab: DisposableDashboardTab, view: azdata.ModelView): DisposableTab {
+		let tab: DisposableTab;
+		if (dashboardTab.toolbar) {
 			const flexContainer = view.modelBuilder.flexContainer().withLayout({ flexFlow: 'column' }).component();
-			flexContainer.addItem(tab.toolbar, { flex: '0 0 auto' });
-			flexContainer.addItem(tab.content, { flex: '1 1 auto' });
-			return {
-				title: tab.title,
-				id: tab.id,
+			flexContainer.addItem(dashboardTab.toolbar, { flex: '0 0 auto' });
+			flexContainer.addItem(dashboardTab.content, { flex: '1 1 auto' });
+			tab = {
+				title: dashboardTab.title,
+				id: dashboardTab.id,
 				content: flexContainer,
-				icon: tab.icon
+				icon: dashboardTab.icon,
+				dispose: () => {
+					dashboardTab.toolbar.dispose();
+					dashboardTab.content.dispose();
+				},
 			};
 		} else {
-			return tab;
+			tab = dashboardTab;
```

Comment:
You still need to create the new object with the dispose function on it here, since the input we get is from extensions and so won't actually have dispose implemented. 

And then the function should be disposing of the content object.